lib/Conversion/TorchToLinalg/DataMovement.cpp

```diff
@@ -1095,6 +1120,79 @@ class ConvertValsemVariantAtenCopyOp
 };
 } // namespace
 
+namespace {
+class ConvertAtenSliceScatterOp
+    : public OpConversionPattern<AtenSliceScatterOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenSliceScatterOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+
+    Location loc = op.getLoc();
+    TypeConverter *typeConverter = getTypeConverter();
+
+    auto input = adaptor.self();
+
+    RankedTensorType inputType = input.getType().cast<RankedTensorType>();
+    RankedTensorType resultType =
+        typeConverter->convertType(op->getResult(0).getType())
+            .cast<RankedTensorType>();
+    Value zero = rewriter.create<arith::ConstantIndexOp>(loc, 0);
+    Value one = rewriter.create<arith::ConstantIndexOp>(loc, 1);
+
+    int64_t dim;
+    if (!matchPattern(op.dim(), m_TorchConstantInt(&dim)))
+      return op->emitError("unimplemented: dim is not constant");
+
+    SmallVector<Value> inputShape = getTensorSizes(rewriter, loc, input);
+    Value dimSize = inputShape[dim];
+
+    Value startTorchType = op.start();
+    Value endTorchType = op.end();
+    Value startBuiltinType = adaptor.start();
+    Value endBuiltinType = adaptor.end();
+
+    if (startTorchType.getType().isa<OptionalType>() ||
+        endTorchType.getType().isa<OptionalType>())
+      return rewriter.notifyMatchFailure(op, "unimplemented optional type arg");
+
+    int64_t step;
+    if (!matchPattern(op.step(), m_TorchConstantInt(&step))) {
+      if (!op.step().getType().isa<Torch::NoneType>())
+        return op->emitError("unimplemented: step is not constant");
+      step = 1;
+    }
+
+    SmallVector<Value> resultShape;
+    SmallVector<Value> offsets(inputType.getRank(), zero);
+    SmallVector<Value> strides(inputType.getRank(), one);
+    prepareArgumentsForSlicingOp(rewriter, loc, resultShape, offsets, strides,
+                                 startTorchType, startBuiltinType, endTorchType,
+                                 endBuiltinType, step, dimSize, dim, input);
```

Comment:
All this code is the same in the conversion patterns of `AtenSliceTensorOp` and `AtenSliceScatterOp`. If you turn the helper `prepareArgumentsForSlicingOp` into a template, you should be able to perform this entire calculation inside that helper function. This will also simplify the type signature quite a bit